#11600 Make generic card more generic

tbrlpld opened the pull request:
# Description

<!-- Describe the PR here -->
This PR updates the generic card template to be more generic. 

Previously it was only possible to pass a page. While this works often it fails in cases where we might want to link to external pages. See #11585. 

This PR adds the option to pass url, image, and title directly to the card template making it more re-usable. It also adds extra blocks for overrides.

Currently, the generic card is used as basis for the blog card (on the blog index) and for the related links in each blog page. 

Link to sample test page: http://localhost:8000/en/blog/, http://localhost:8000/en/blog/initial-test-blog-post-with-fixed-title/
Related PRs/issues: #11600

# Checklist

<!-- Check off items with `[x]` or cross out items that don't apply with `~~The description~~` -->

**Tests**
- [x] Is the code I'm adding covered by tests?

**Changes in Models:**
- [-] Did I update or add new fake data?
- [-] Did I squash my migration?
- [x] Are my changes [backward-compatible](https://github.com/mozilla/foundation.mozilla.org/blob/main/docs/workflow.md#django-migrations-what-to-do-when-working-on-backward-incompatible-migrations). If not, did I schedule a deploy with the rest of the team?

**Documentation:**
- [-] Is my code documented?
- [-] Did I update the READMEs or wagtail documentation?

**Merge Method**
**💡❗Remember to use squash merge when merging non-feature branches into `main`**

danielfmiranda reviewed (changes requested):
Hi @tbrlpld thanks for taking this work on! the changes to `generic_card.html` look good to me. 

However, it appears that there may be some [unintended changes/commits in the history](https://github.com/MozillaFoundation/foundation.mozilla.org/pull/11601/commits/67365e10fea5dcc0b6b46cd3c2e10170a6cd23dd). 

Just wanted to double check if this change was intended and if not, would it be possible to remove it so we can merge in the intended changes?

Thank you!

tbrlpld commented:
> Hi @tbrlpld thanks for taking this work on! the changes to `generic_card.html` look good to me.
> 
> However, it appears that there may be some [unintended changes/commits in the history](https://github.com/MozillaFoundation/foundation.mozilla.org/pull/11601/commits/67365e10fea5dcc0b6b46cd3c2e10170a6cd23dd).
> 
> Just wanted to double check if this change was intended and if not, would it be possible to remove it so we can merge in the intended changes?
> 
> Thank you!

Ah good catch! Thanks @danielfmiranda. I have rebased and dropped the commit in question.
